File: construction_management/overrides/daily_roster.py
# Copyright (c) 2026, Construction Management and contributors

import frappe


def set_project_short_name(doc, method=None):
	if not doc.project:
		doc.custom_project_short_name = None
		return

	short_name = frappe.db.get_value("Project", doc.project, "custom_project_short_name")
	if short_name:
		doc.custom_project_short_name = short_name

File: construction_management/public/js/daily_roster.js
frappe.ui.form.on("Daily Roster", {
	project(frm) {
		if (!frm.doc.project) {
			frm.set_value("custom_project_short_name", "");
			return;
		}

		frappe.db.get_value(
			"Project",
			frm.doc.project,
			"custom_project_short_name",
			(r) => {
				frm.set_value("custom_project_short_name", r?.custom_project_short_name || "");
			}
		);
	},
});
